src/frontend/src/storage/index.test.ts

```diff
@@ -144,33 +205,64 @@ test(
  * If `after` is a function, the function is called with the content of the storage.
  */
 function withStorage(
-  fn: () => void,
+  fn: () => void | Promise<void>,
   opts?: {
     localStorage?: {
       before?: LocalStorage;
       after?: LocalStorage | ((storage: LocalStorage) => void);
     };
+    indexeddb?: {
+      before?: IndexedDB;
+      after?: IndexedDB | ((storage: IndexedDB) => void);
+    };
   }
-): () => void {
-  return () => {
+): () => Promise<void> {
+  return async () => {
     localStorage.clear();
-    const before = opts?.localStorage?.before;
-    if (nonNullish(before)) {
-      setLocalStorage(before);
+    const lsBefore = opts?.localStorage?.before;
+    if (nonNullish(lsBefore)) {
+      setLocalStorage(lsBefore);
     }
-    fn();
-    const after = opts?.localStorage?.after;
-    if (nonNullish(after)) {
-      const actual: LocalStorage = readLocalStorage();
 
-      if (typeof after === "function") {
-        after(actual);
+    await idb.delMany(await idb.keys());
```

Comment:
- `clear();` would be more handy?
- you don't use a custom store (general question)?

Reply:
> would be more handy?

Just missed it, thanks!

> you don't use a custom store (general question)?

no, just using the default `keyval-store` DB and `keyval` store